js/ui/form/ui.form.layout_manager.js

```diff
@@ -141,6 +141,10 @@ var LayoutManager = Widget.inherit({
         var layoutData = this.option("layoutData"),
             newValue = value;
 
+        if(!typeUtils.isDefined(layoutData)) {
```

Comment:
If you set an item with the `dataField` option you cannot update undefined `formData` by changing editor's value. I suggest set an empty object to the `formData` on attempt change it to null or undefined.

By the way don't forget about the [T597477](https://www.devexpress.com/Support/Center/Question/Details/T597477/dxform-values-not-being-set-when-form-data-isn-t-defined) ticket